---
title: "features_as_function_of_reads"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
list.of.packages <- c("tidyverse")
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
if(length(new.packages)) install.packages(new.packages)
lapply(list.of.packages, require, character.only = TRUE)
```


```{r}
#META <- '/pita/users/hila/analysis/datasets/hmp2_metadata_mgx_no_na_cols_w_bakery_results.tsv'
#META <- '/pita/users/hila/analysis/datasets/hmp2_metadata_mgx_mtx.tsv'
```


```{r}
meta.df <- read.delim(META, stringsAsFactors = F, check.names = T)
head(meta.df)
```

```{r}
last_col_names <- meta.df %>% colnames() %>% tail(n=8)
last_col_names
```

```{r}
str <- "pathways"
#str <- "enzymes"
#str <- "gene families"
#str <- "MetaPhlAn species"
title <- sprintf("Number of %s per sample\nas a function of metagenome reads", str)
p <- meta.df %>% ggplot(aes(x = final, y = humann_pathabundance_relab_counts, color = diagnosis)) +
  geom_point() +
  scale_x_continuous(labels = scales::comma, breaks = seq(0,28000000,5000000)) +
  ggtitle(title) +
  labs(x = "# of filtered reads by KneadData (forward secquence)", 
       y = paste("# of", str, "per sample")) +
  theme(text = element_text(size = 12),
        plot.title = element_text(hjust = 0.5, size = 16)) 
p
```


```{r}
png(sprintf("/pita/users/hila/analysis/plots/%s.png", title), width = 600, height = 400)
p
dev.off()
```
